internal/cli/console.go

```diff
@@ -201,15 +213,59 @@ func (o *ConsoleOptions) createSessionMetadata(t k8sTerm.TTY, passThroughArgs []
 	return string(b)
 }
 
-func (o *ConsoleOptions) connectViaWS(ctx context.Context, config *client.Config, deviceName, token string, passThroughArgs []string) error {
+type closeOnTildeDotReader struct {
+	r                io.Reader
+	tildeEncountered bool
+	eofDetected      bool
+	cancel           context.CancelFunc
+}
+
+func newCloseOnTildeDotReader(inner io.Reader, cancel context.CancelFunc) *closeOnTildeDotReader {
+	return &closeOnTildeDotReader{
+		r:      inner,
+		cancel: cancel,
+	}
+}
+
+func (e *closeOnTildeDotReader) Read(p []byte) (int, error) {
+	if e.eofDetected {
+		e.cancel()
+		return 0, io.EOF
+	}
 
+	n, err := e.r.Read(p)
+
+	for i := 0; i < n; i++ {
+		switch p[i] {
+		case '~':
```

Comment:
It should only be after a newline, like in SSH

Reply:
Changed